### ISSUE TITLE
-Wreserved-identifier should not warn on system macros

### ISSUE THREAD

**Issue by LebedevRI** [llvm/llvm-project#49592]
|  |  |
| --- | --- |
| Bugzilla Link | [50248](https://llvm.org/bz50248) |
| Resolution | FIXED |
| Resolved on | Jun 10, 2021 01:38 |
| Version | unspecified |
| OS | Linux |
| CC | @AaronBallman,@zygoloid,@serge-sans-paille |
| Fixed by commit(s) | 6045cb89e5e8223eea4d7f9c3a3e0d8d25ff909f |

## Extended Description 
It currently broke google benchmark/test, even when their headers are pulled in via -isystem.

**Comment by firewave** [comment]
This is still an issue with system headers.

```cpp
#include <sys/select.h>

static void f()
{
    fd_set rfds;
    FD_ZERO(&rfds);
}
```

reports 

```
<source>:6:5: warning: identifier '__d0' is reserved because it starts with '__' [-Wreserved-identifier]
    FD_ZERO(&rfds);
    ^
/usr/include/x86_64-linux-gnu/sys/select.h:88:26: note: expanded from macro 'FD_ZERO'
#define FD_ZERO(fdsetp)         __FD_ZERO (fdsetp)
                                ^
/usr/include/x86_64-linux-gnu/bits/select.h:35:9: note: expanded from macro '__FD_ZERO'
    int __d0, __d1;                                                           \
        ^
<source>:6:5: warning: identifier '__d1' is reserved because it starts with '__' [-Wreserved-identifier]
/usr/include/x86_64-linux-gnu/sys/select.h:88:26: note: expanded from macro 'FD_ZERO'
#define FD_ZERO(fdsetp)         __FD_ZERO (fdsetp)
                                ^
/usr/include/x86_64-linux-gnu/bits/select.h:35:15: note: expanded from macro '__FD_ZERO'
    int __d0, __d1;                                                           \
```

See https://godbolt.org/z/heszsbEjx.

It also happens with the `QBENCHMARK` from Qt.

**Comment by serge-sans-paille** [comment]
I can reproduce, reopening the issue to keep track of the bug.

**Comment by serge-sans-paille** [comment]
Should be fixed by https://reviews.llvm.org/D118532
